docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,107 @@
+# Modifications to Fluentd Metadata Retrieval Proposal
+
+##Motivation
+In the current design (v3.4) of the EFK stack, there is a Fluentd pod deployed to each Openshift node. Fluentd is responsible for reading container logs, adding pod metadata (e.g. labels, annotations) to individual log entries, and sending the logs to the Elasticsearch cluster. The additional metadata, such as labels, augment log entries to allow users to intelligently query their deployed services.  The metadata acts to enrich the logs so users may logically group their services as a single, coherent application.
+
+The pod metadata is currently sourced by watching the Openshift API server for Pod information via the [Kubernetes Metadata Filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter)[2]; Each watch maintains an open connection to the API server. Testing at larger scales of Openshift clusters has shown issues where Fluentd is unable to connect to the API server because it is limited by the **maxRequestsInFlight** configuration setting [1].  The current deployment topology at scale:
+
+* Results in logs not being shipped to the aggregator and thus being unavailable
+* Limits the number of cluster nodes because of the cap on the number of open connections
+
+This proposal addresses both the short and long term solutions to resolve the connection issues being experienced using the v3.4 EFK stack.
+### Considerations
+1. OpenShift Logging is not the only consumer of OpenShift log messages.  There are customers who desire to use their existing logging infrastructure(e.g. Splunk, Elasticsearch).  Log entries need to be enriched with metadata irrespective of the final destination.
+2. It is speculated that there is a low probability that labels and annotations will change during the lifetime of a running pod. We should determine if 'stale' labels and annotations are acceptable
+3. Identify the tolerance for dropped message.
+
+## Proposal - Short Term (3.4)
+1. Leave the value as-is and note that it must be adjusted to account for # of nodes plus some headroom (swag @ 10%?).
+1. Increase the default to our current maximum supported node count, plus some headroom.
+1. Increasing maxRequestsInflight upstream
+1. **COMPLETE** ~~Modify Fluentd Metadata plugin to cache by namespace and pod name only~~[2][4]
+
+## Proposal - Long Term
+### Option 1 - Admission Controller
+1. Create an admission controller which embeds the necessary metadata as envirnment variables into the pods on submission.  In this way every pod that is run on the system will have its data locally.
```

Comment:
@smarterclayton, do you mean to say that it is unlikely that every pod will have data locally using this method?  Won't we know ahead of time if `fluentd` needs more info?